server/paymentProviders/PaymentProvider.ts

```diff
@@ -0,0 +1,15 @@
+import models from '../models';
+
+import { PaymentProviderService } from './PaymentProviderService';
+
+export interface PaymentProvider {
+  /**
+   * Triggers the payment for this order and updates it accordingly
+   */
+  processOrder(order: typeof models.Order): Promise<typeof models.Transaction>;
+
+  /**
+   * The different types of payment methods supported by this provider
+   */
+  types: Record<string, PaymentProviderService>;
+}
```

Comment:
No strong opinions here either, perhaps unite both these interface files in `paymentProviders/types.ts`?
Moving forward, for internal types, I think it would be interesting to have a single declaration file where we could use its types without needing to explicitly import them. It's a bit of magic but I would think it would be nice to keep types optional.

Reply:
According to https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts, it looks the Typescript team uses `types.ts` files, located in the business unit they correspond to - which matches what you're proposing with `paymentProviders/types.ts`. I will make the change in that direction.

I'm not sure about the part `without needing to explicitly import them`, do you mean to have them in the global scope?